base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -235,13 +238,20 @@ impl<T> BlockchainDatabase<T>
 where T: BlockchainBackend
 {
     /// Creates a new `BlockchainDatabase` using the provided backend.
-    pub fn new(db: T) -> Result<Self, ChainStorageError> {
+    pub fn new(db: T, consensus_manager: &ConsensusManager<T>) -> Result<Self, ChainStorageError> {
```

Comment:
swop to an owned move, as ConsensusManager is an arc, we can cheaply clone. 
If we want DB to keep a copy later on we need to modify this to not hide the clone

Reply:
See previous comment.